src/main/java/org/spongepowered/common/event/filter/delegate/ContextValueFilterSourceDelegate.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.event.filter.delegate;
+
+import static org.objectweb.asm.Opcodes.ACONST_NULL;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ARETURN;
+import static org.objectweb.asm.Opcodes.CHECKCAST;
+import static org.objectweb.asm.Opcodes.DUP;
+import static org.objectweb.asm.Opcodes.GETFIELD;
+import static org.objectweb.asm.Opcodes.GOTO;
+import static org.objectweb.asm.Opcodes.IFEQ;
+import static org.objectweb.asm.Opcodes.IFNE;
+import static org.objectweb.asm.Opcodes.INSTANCEOF;
+import static org.objectweb.asm.Opcodes.INVOKEINTERFACE;
+import static org.objectweb.asm.Opcodes.INVOKESTATIC;
+import static org.objectweb.asm.Opcodes.INVOKEVIRTUAL;
+import static org.objectweb.asm.Opcodes.PUTFIELD;
+
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
+import org.spongepowered.api.GameRegistry;
+import org.spongepowered.api.event.cause.EventContext;
+import org.spongepowered.api.event.cause.EventContextKey;
+import org.spongepowered.api.event.filter.cause.ContextValue;
+import org.spongepowered.common.SpongeImpl;
+import org.spongepowered.common.registry.SpongeGameRegistry;
+
+import java.lang.reflect.Parameter;
+import java.util.Optional;
+
+public class ContextValueFilterSourceDelegate extends ContextFilterSourceDelegate {
+
+    private final ContextValue anno;
+
+    public ContextValueFilterSourceDelegate(ContextValue anno) {
+        this.anno = anno;
+
+        Optional<EventContextKey> opKey = SpongeImpl.getRegistry().getType(EventContextKey.class, this.anno.value());
```

Comment:
Does this even work? I don't believe the context keys are I'd with the way that the annotation is intending. It was intended to modify the annotation to improve it, since the class type filter is easily remedied by the declaration type of the method parameter of the listener.

Does the test pass? Can you verify a test plugin will function given the test?

Reply:
Yeah it works and the test passes, I don't really understand the problem, how was it intended to be improved?

Reply:
Because, the id's are not the same as the field names found in `ContextValueKeys` class.

Reply:
I don't really understand why wouldn't the ids be the same, but well, how am I supposed to change that?

Reply:
? 🤔

Reply:
You can't. We have to change the API as it's currently unimplementable. If you want to change the API to do it, by all means, you'd need some sort of class reference (to look up where the thing comes from) and the field name. That would be a way to do it.